### PR TITLE
Dfr 3080

### PR DESCRIPTION
**Pull request by ptrelease**
### Jira link

https://tools.hmcts.net/jira/browse/DFR-3080

### Change description

RefugeWrapper added with two fields for applicant in refuge.  Idea being that applicantInRefugeTab is populated from applicantInRefugeQuestion in the SolicitorCreateContestedAboutToSubmitHandler.

### Testing done

Manual testing, when confirming CCD definitions correct.
Explicit unit tests.

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateContestedAboutToSubmitHandlerTest.java [hunk 1]
```diff
@@ -141,6 +141,37 @@ public void givenCase_whenMandatoryDataValidationFails_thenReturnsErrors() {
         Assertions.assertThat(response.getData()).isNotNull();
     }
 
+    @Test
+    public void givenCase_whenApplicantRefugeQuestionAnswered_thenApplicantRefugeTabUpdated() {
```
Comment (so99y) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateContestedAboutToSubmitHandlerTest.java:145]: It is worth taking this opportunity to upgrade the test to JUnit 5.

Change the annotation of this class to `@ExtendWith(MockitoExtension.class)` and do the refactoring.

Comment (ptrelease) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateContestedAboutToSubmitHandlerTest.java]: Thanks.  Refactored.

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateContestedAboutToSubmitHandler.java [hunk 2]
```diff
@@ -82,6 +83,15 @@ public GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle(Finrem
 
         representationWorkflowService.persistDefaultOrganisationPolicy(caseData);
 
+        // updates the tab version of applicantInRefuge with the question version, then removes question version.
+        Optional.ofNullable(caseData.getRefugeWrapper().getApplicantInRefugeQuestion())
+                .ifPresent(question -> {
+                    caseData.getRefugeWrapper().setApplicantInRefugeTab(question);
+                    log.info("Updating applicantInRefugeTab for case reference {}. Removing applicantInRefugeQuestion",
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateContestedAboutToSubmitHandler.java:90]: I question whether this log message is of any value to us. Logs do actually cost money.

**Review by jthmcts — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateContestedAboutToSubmitHandler.java (hunk 2 above)
Comment (ptrelease) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/solicitorcreatecase/SolicitorCreateContestedAboutToSubmitHandler.java:90]: @jthmcts Thanks for raising.  Will contact you to discuss.

**Review by so99y — APPROVED** [review]

**Review by jthmcts — APPROVED** [review]

**Review by so99y — APPROVED** [review]

**Review by jthmcts — APPROVED** [review]

**Pull request merged**